### DISCUSSION TITLE
precommit: pkg/config/loader.go:8:2: package slices is not in GOROOT

### DISCUSSION THREAD

**Discussion by stdedos** [golangci/golangci-lint#4830] [Q&A] [answered]
https://github.com/stdedos/junit2html/actions/runs/9584573390/job/26428499894

```
Error:     pkg/config/loader.go:8:2: package slices is not in GOROOT (/opt/hostedtoolcache/go/1.19.13/x64/src/slices)
```

But then I see

```
[INFO] Installing environment for https://github.com/golangci/golangci-lint.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/opt/hostedtoolcache/go/1.19.13/x64/bin/go', 'install', './...')
```

... I thought `golangci-lint` was "not supposed to be installed", right? 😕

**Comment by ldez** [comment]
You are using go1.19.13, `slices` required go1.21.

You are trying to compile golangci-lint.

**Reply by stdedos** [reply]
> You are trying to compile golangci-lint.

Yeah, I am not 😅 But for some reason, running "pre-commit-ci-lite" works like that 🤷 
... hopefully locally-running `pre-commit run -a` does not work _similarly_, and instead it uses the pre-compiled binary

Or maybe I should inject it to the CI environment? (`curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.59.1`)

**Reply by ldez** [reply]
I don't know how pre-commit works so I cannot help more.

I can just recommend the usage of our GitHub action: https://golangci-lint.run/welcome/install/#github-actions

**Reply by stdedos** [reply, marked as answer]
Maybe "something weird" happens on version mismatch.

Using go 1.21 works just fine https://github.com/stdedos/junit2html/actions/runs/9585149784/job/26430376871 **and** does not seem to be re-building. (There is a slight _caching_ possibility, but "I don't think so").

**Reply by ldez** [reply]
>  does not seem to be re-building.

It was not a "re-build", but just a build because the precommit hook doesn't install the binary.
And you still have a build but the logs of this build are hidden.

> Maybe "something weird" happens on version mismatch.

You can use golangci-lint with go1.19 but you should compile it with go1.21.

There's nothing weird here, just the expected behavior related to compilation without the right Go version.
